kp.go

```diff
@@ -256,6 +257,12 @@ func (c *Client) do(ctx context.Context, req *http.Request, res interface{}) (*h
 		req.Header.Set("x-kms-key-ring", c.Config.KeyRing)
 	}
 
+	if c.Config.Headers != nil {
```

Comment:
I would suggest using length instead of nil to check if Headers(map) is empty or not.
```suggestion
	if len(c.Config.Headers) != 0 {
```